### PR TITLE
worker:API to get config JSON object

### PR DESCRIPTION
**Pull request by PriyangaRamasamy**
Method to get system config JSON object which got created in
worker class.

### CODE REVIEW

File: include/worker.hpp [hunk 1]
```diff
@@ -100,6 +100,16 @@ class Worker
                       types::ObjectMap& objectInterfaceMap,
                       const std::string& vpdFilePath);
 
+    /**
```
Comment (jinuthomas) [inline include/worker.hpp:103]: Could you please remind me why this was needed ?

File: include/worker.hpp [hunk 2]
```diff
@@ -100,6 +100,16 @@ class Worker
...
+     * @brief API to get VPD config JSON object
+     *
+     * @return parsed VPD config JSON object
```
Comment (SunnySrivastava1984) [inline include/worker.hpp:106]: > VPD

Not a VPD config JSON, it is system config JSON.

File: include/worker.hpp [hunk 3]
```diff
@@ -100,6 +100,16 @@ class Worker
...
+     */
+    inline nlohmann::json getConfig() const
```
Comment (SunnySrivastava1984) [inline include/worker.hpp:108]: > getConfig

Be more specific, getParsedSystemConfig or something like that.

**Review by SunnySrivastava1984 — CHANGES_REQUESTED** [review]

File: include/worker.hpp (hunk 1 above)
Comment (PriyangaRamasamy) [reply include/worker.hpp:103]: we decided not to create n copies of json object. so always get from worker class.

File: include/worker.hpp (hunk 2 above)
Comment (PriyangaRamasamy) [reply include/worker.hpp]: done

File: include/worker.hpp (hunk 3 above)
Comment (PriyangaRamasamy) [reply include/worker.hpp]: done

**Review by SunnySrivastava1984 — CHANGES_REQUESTED** [review]
No commit message?

File: include/worker.hpp [hunk 4]
```diff
@@ -100,6 +100,16 @@ class Worker
                       types::ObjectMap& objectInterfaceMap,
                       const std::string& vpdFilePath);
 
+    /**
+     * @brief API to get system config JSON object
+     *
+     * @return parsed system config JSON object.
+     */
+    inline nlohmann::json getSysConfigJSONObj() const
```
Comment (SunnySrivastava1984) [inline include/worker.hpp:108]: > getSysConfigJSONObj

getSysConfigJsonObj

**Review by SunnySrivastava1984 — CHANGES_REQUESTED** [review]

File: include/worker.hpp [hunk 5]
```diff
@@ -100,6 +100,16 @@ class Worker
                       types::ObjectMap& objectInterfaceMap,
                       const std::string& vpdFilePath);
 
+    /**
+     * @brief API to get system config JSON object
+     *
+     * @return parsed system config JSON object.
```
Comment (jinuthomas) [inline include/worker.hpp:106]: nit: remove parsed

Comment (PriyangaRamasamy) [reply include/worker.hpp]: done

File: include/worker.hpp (hunk 4 above)
Comment (PriyangaRamasamy) [reply include/worker.hpp]: done

**Review by SunnySrivastava1984 — APPROVED** [review]

**Review by jinuthomas — APPROVED** [review]

File: include/worker.hpp [hunk 6]
```diff
@@ -110,6 +110,16 @@ class Worker
         return m_isAllFruCollected;
     }
 
+    /**
+     * @brief API to get system config JSON object
+     *
+     * @return System config JSON object.
+     */
+    inline nlohmann::json getSysConfigJsonObj() const
```
Comment (jinuthomas) [inline include/worker.hpp:118]: why getSysConfigJsonObj and not getSysCfgJsonObj

**Review by jinuthomas — APPROVED** [review]

File: include/worker.hpp (hunk 6 above)
Comment (PriyangaRamasamy) [reply include/worker.hpp:118]: no harm in expanding a word in function name. it gives better readability only.
we discussed to shorten the names only for variables as variables will be used in multiple lines within the same api which increases the number of lines.

Comment (PriyangaRamasamy) [reply include/worker.hpp:118]: done

**Review by jinuthomas — APPROVED** [review]

**Pull request merged**